pkg/validating-webhooks/agentinstall/v1beta1/infraenv_admission_hook.go

```diff
@@ -0,0 +1,194 @@
+package v1beta1
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/openshift/assisted-service/api/v1beta1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	log "github.com/sirupsen/logrus"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	infraEnvGroup    = "agent-install.openshift.io"
+	infraEnvVersion  = "v1beta1"
+	infraEnvResource = "infraenvs"
+
+	infraEnvAdmissionGroup   = "admission.agentinstall.openshift.io"
+	infraEnvAdmissionVersion = "v1"
+)
+
+// InfraEnvValidatingAdmissionHook is a struct that is used to reference what code should be run by the generic-admission-server.
+type InfraEnvValidatingAdmissionHook struct {
+	decoder *admission.Decoder
+}
+
+// NewInfraEnvValidatingAdmissionHook constructs a new AgentClusterInstallValidatingAdmissionHook
+func NewInfraEnvValidatingAdmissionHook(decoder *admission.Decoder) *InfraEnvValidatingAdmissionHook {
+	return &InfraEnvValidatingAdmissionHook{decoder: decoder}
+}
+
+// ValidatingResource is called by generic-admission-server on startup to register the returned REST resource through which the
+//                    webhook is accessed by the kube apiserver.
+// For example, generic-admission-server uses the data below to register the webhook on the REST resource "/apis/admission.agentinstall.openshift.io/v1/agentclusterinstalls".
```

Comment:
"/apis/admission.agentinstall.openshift.io/v1/agentclusterinstalls" -> "/apis/admission.agentinstall.openshift.io/v1/infraenvvalidators"

You can fix also the comment in the agentclusterinstall_admission_hook.go comment